core/src/main/java/feast/core/service/JobService.java

```diff
@@ -158,16 +158,15 @@ public RestartIngestionJobResponse restartJob(RestartIngestionJobRequest request
     // check job exists
     Optional<Job> getJob = this.jobRepository.findById(request.getId());
     if (getJob.isEmpty()) {
+      // FIXME: if getJob.isEmpty then constructing this error message will always throw an error...
       throw new NoSuchElementException(
           "Attempted to stop nonexistent job with id: " + getJob.get().getId());
```

Comment:
This is bug, I'll raise it in a separate issue/PR, didn't want to change it here because it should probably have a hotfix instead of waiting on this refactoring PR to get through. There are two or three instances of the same problem in this file.

The intention here was probably to use `request.getId()` for the exception message. `getJob.get()` will throw `NoSuchElementException` by definition. @mrzzy

Reply:
Good catch. It should be `request.getId()` instead.